### PR TITLE
Create spans for vulnerabilities outside of requests 

### PR DESCRIPTION
**Pull request by CarlesDD**
### What does this PR do?
Creates a new span for vulnerabilities where there is no current context

### Motivation
Vulnerabilities are not only found during the execution of a request, they can be present at boot time, background processes ... this PR tries to enable the creation of custom spans when there is no requests.

### CODE REVIEW

File: packages/dd-trace/src/appsec/iast/index.js [hunk 1]
```diff
@@ -14,17 +14,20 @@ const IAST_ENABLED_TAG_KEY = '_dd.iast.enabled'
 const requestStart = dc.channel('dd-trace:incomingHttpRequestStart')
 const requestClose = dc.channel('dd-trace:incomingHttpRequestEnd')
 
-function enable (config) {
+function enable (config, _tracer) {
```
Comment (simon-id) [inline packages/dd-trace/src/appsec/iast/index.js:17]: You can access the tracer in a global var, something like `global._ddtrace.tracer` if i'm not wrong. I'm not sure which approach is best but at least now you know :+1:

File: packages/dd-trace/src/appsec/iast/overhead-controller.js [hunk 2]
```diff
@@ -80,11 +83,27 @@ function configure (cfg) {
...
+  _resetGlobalContext()
+  resetGlobalContextInterval = setInterval(() => {
+    _resetGlobalContext()
+  }, INTERVAL_RESET_GLOBAL_CONTEXT)
+}
+
+function finishGlobalContext () {
+  if (resetGlobalContextInterval) {
+    clearInterval(resetGlobalContextInterval)
+    resetGlobalContextInterval.unref && resetGlobalContextInterval.unref()
```
Comment (uurien) [inline packages/dd-trace/src/appsec/iast/overhead-controller.js:97]: You should call to `unref` just after `setInterval`, not in `finishGlobalContext` that is called only from the test.

File: packages/dd-trace/src/appsec/iast/index.js (hunk 1 above)
Comment (uurien) [reply packages/dd-trace/src/appsec/iast/index.js:17]: I think that you can also require the tracer instance, it is exported here: https://github.com/DataDog/dd-trace-js/blob/7fb068aff5a877360e968523935811832ae7a75f/packages/dd-trace/index.js#L17

**Review by uurien — CHANGES_REQUESTED** [review]
I miss some tests here:
https://github.com/DataDog/dd-trace-js/blob/7fb068aff5a877360e968523935811832ae7a75f/packages/dd-trace/test/appsec/iast/analyzers/weak-hash-analyzer.spec.js#L64
To check that the analyzer is working correctly out of the request and sending spans correctly.

**Review by Qard — DISMISSED** [review]

File: packages/dd-trace/src/appsec/iast/index.js (hunk 1 above)
Comment (simon-id) [reply packages/dd-trace/src/appsec/iast/index.js:17]: whoops ok, nevermind then

**Review by uurien — APPROVED** [review]

**Review by Qard — APPROVED** [review]

**Pull request merged**